pkg/controllers/management/globaldns/globaldns_handler.go

```diff
@@ -0,0 +1,228 @@
+package globaldns
+
+import (
+	"context"
+	"net"
+	"sort"
+
+	v1beta1Rancher "github.com/rancher/types/apis/extensions/v1beta1"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	clientv1beta1 "k8s.io/client-go/kubernetes/typed/extensions/v1beta1"
+)
+
+const (
+	GlobaldnsController = "mgmt-global-dns-controller"
+)
+
+type GlobalDNSController struct {
+	globalDNSInterface v3.GlobalDNSInterface
+	globalDNSLister    v3.GlobalDNSLister
+	//ingressInterface   clientv1beta1.IngressInterface //need to use client-go IngressInterface to update Ingress.Status field
+	ingressLister     v1beta1Rancher.IngressLister
+	ctx               context.Context
+	managementContext *config.ManagementContext
```

Comment:
it's better we don't store the context in the struct; just initialize and store the explicit lister or interface instead